travis.yml

```diff
@@ -0,0 +1,66 @@
+# use Java and non-root
+dist: trusty
+sudo: false
+language: Java
+jdk: oraclejdk8
+
+# ignore the virtualenv that Travis creates
+env:
+  global:
+    - PIPENV_VENV_IN_PROJECT=1
+    - PIPENV_IGNORE_VIRTUALENVS=1
+
+# install the dependencies
+before_install:
+  # install mdl for linting
+  - gem install mdl
+  # download gradle as a zip file
+  - wget https://services.gradle.org/distributions/gradle-4.6-bin.zip
+  - unzip -d $HOME gradle-4.6-bin.zip
+  # delete the downloaded file
+  - rm -rf gradle-4.6-bin.zip
+  # add gradle bin to path at the beginning to ensure it overwrites old gradle
+  - export PATH="$HOME/gradle-4.6/bin:$PATH"
+  # switch to Python 3.6.3 globally, install python dependencies
+  - pyenv global 3.6.3
+  - pip install --upgrade pip
+  - pip install proselint
+  - pip install pipenv
+  # do not allow Travis to use standard error for debugging purposes
+  - unset _JAVA_OPTIONS
+  # install dev dependencies with pipenv
+  - pipenv install --dev --three
```

Comment:
I don't think we need `--three` here. You can put the commands in the `README` in `travis.yml` for now